tests/components/ipma/test_config_flow.py

```diff
@@ -116,3 +121,53 @@ async def test_flow_entry_config_entry_already_exists():
         assert len(config_form.mock_calls) == 1
         assert len(config_entries.mock_calls) == 1
         assert len(flow._errors) == 1
+
+
+async def test_config_entry_migration(hass):
+    """Tests config entry without mode in unique_id can be migrated."""
+    ipma_entry = MockConfigEntry(
+        domain=DOMAIN,
+        title="Home",
+        data={CONF_LATITUDE: 0, CONF_LONGITUDE: 0, CONF_MODE: "daily"},
+    )
+    ipma_entry.add_to_hass(hass)
+
+    ipma_entry2 = MockConfigEntry(
+        domain=DOMAIN,
+        title="Home",
+        data={CONF_LATITUDE: 0, CONF_LONGITUDE: 0, CONF_MODE: "hourly"},
+    )
+    ipma_entry2.add_to_hass(hass)
+
+    mock_registry(
+        hass,
+        {
+            "weather.hometown": entity_registry.RegistryEntry(
+                entity_id="weather.hometown",
+                unique_id="0, 0",
+                platform="ipma",
+                config_entry_id=ipma_entry.entry_id,
+            ),
+            "weather.hometown_2": entity_registry.RegistryEntry(
+                entity_id="weather.hometown_2",
+                unique_id="0, 0, hourly",
+                platform="ipma",
+                config_entry_id=ipma_entry.entry_id,
+            ),
+        },
+    )
+
+    with patch(
+        "homeassistant.components.ipma.weather.async_get_location",
+        return_value=MockLocation(),
+    ):
+        assert await async_setup_component(hass, DOMAIN, {})
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)